--- src/aui/slbook.cpp.orig	2013-07-21 15:31:23.957317554 +0200
+++ src/aui/slbook.cpp	2013-07-21 15:37:38.502656620 +0200
@@ -323,8 +323,7 @@
 
 		if (p) tabs += wxT(",");
 
-		if ((int)page_idx == m_curpage) tabs += wxT("*");
-		else if ((int)p == tabframe->m_tabs->GetActivePage()) tabs += wxT("+");
+		if ((int)p == tabframe->m_tabs->GetActivePage()) tabs += wxT("+");
 		tabs += (wxFormat(wxT("%u") ) % page_idx).str();
      }
   }
@@ -370,13 +369,14 @@
 
      // create a new tab frame
      wxTabFrame* new_tabs = new wxTabFrame;
+     static int m_tab_id_counter = 0;
      new_tabs->m_tabs = new wxAuiTabCtrl(this,
                                 m_tab_id_counter++,
                                 wxDefaultPosition,
                                 wxDefaultSize,
                                 wxNO_BORDER|wxWANTS_CHARS);
      new_tabs->m_tabs->SetArtProvider(m_tabs.GetArtProvider()->Clone());
-     new_tabs->SetTabCtrlHeight(m_tab_ctrl_height);
+     new_tabs->SetTabCtrlHeight(new_tabs->m_tab_ctrl_height);
      new_tabs->m_tabs->SetFlags(m_flags);
      wxAuiTabCtrl *dest_tabs = new_tabs->m_tabs;
 
@@ -418,7 +418,6 @@
   m_mgr.LoadPerspective(frames);
 
   // Force refresh of selection
-  m_curpage = -1;
   SetSelection(sel_page);
 
   return true;
